tlatools/org.lamport.tlatools/src/util/SimpleFilenameToStream.java

```diff
@@ -136,112 +118,94 @@ private String[] getLibraryPaths(final String installationBasePath, String[] lib
-    if (path == null) {
-      res = new String[1];
-      res[0] = installationBasePath + FileUtil.separator;
-    }
-    else {
+    if (path != null) {
       String[] paths = path.split(FileUtil.pathSeparator);
-      res = new String[paths.length+1];
-      for(int i=0;i<paths.length;i++){
-	res[i] = paths[i];
-	if (!res[i].endsWith(FileUtil.separator)){
-	  res[i] = res[i] + FileUtil.separator;
-	}
+      for (String s : paths) {
+          res.add(new FilesystemResolver(Path.of(s)));
       }
-      res[paths.length] = installationBasePath + FileUtil.separator;
     }
-    return res;
+
+    // Finally, search the bare classpath (e.g. for the community modules)
+    res.add(new ClasspathResolver(cl, ""));
+
+    return new SequentialResolver(res);
   }
 
   /**
-   *  From name, prepend path elements to try to find the actual
-   *  file intended, and use the resulting fully-qualified name to initialize
-   *  the File.  The first fully-qualified name that refers to a file that actually
-   *  exists is the File returned; but if none exists, the last one tried is returned
+   *  From name, try to find the actual file intended.  If the file is not a normal
+   *  file (for instance, if it is an entry in a JAR file), then copy it to the
+   *  filesystem and return the copied file.
+   *
+   *  <p>The first fully-qualified name that refers to a file that actually
+   *  exists is the File returned; but if none exists, a non-existent file is returned
    *  anyway.  Hence, the File object returned does not necessarily represent a file
    *  that actually exists in the file system.
    *
-   *  @param  module name, used as basis of path name to the file that should contain it
+   *  @param name the name of the file to look for
    */
-  private final File locate(String name)
+  private final TLAFile locate(String name)
   {
+    Optional<TLAFile> userModule = locateUsingResolver(userFileResolver, name, false);
+    if (userModule.isPresent()) {
+        return userModule.get();
+    }
 
-    String prefix      = "";                // directory to be prepended to file name before lookup
-
-    File   sourceFile  = null;              // File object from fully-qualified name of file found by
-                                            //   searching "libraryPath" for "name", as is done for PATHs and CLASSPATHs
+    Optional<TLAFile> libraryModule = locateUsingResolver(standardLibraryResolver, name, true);
+    if (libraryModule.isPresent()) {
+        return libraryModule.get();
+    }
 
-    // The TLA standard modules are presumed to be in a directory tla2sany/StandardModules
-    // i.e. a toplevel subdirectory of the main installation directory for SANY
-    // We append the StandardModules director to the library path, so the user does
-    // not have to do anything special and the standard modules are automatically accessible
+    // Small bit of awkwardness here: this method requires us to return a nonexistent file if the requested one
```

Comment:
Add note if this is for backwards compatibility reasons or what. Can this be fixed at the callsite?

Reply:
I once tried to change the contract here (file-that-might-or-might-not-exist -> `Optional<File>`). There are many callsites to change, and some of them are in the Toolbox.

I will add a note that this is for historical reasons. 👍 For now, at least the contract is well-documented in `FilenameToStream`.